app/uk/gov/hmrc/tai/service/TaxCodeChangeService.scala

```diff
@@ -124,19 +124,29 @@ class TaxCodeChangeServiceImpl @Inject()(taxCodeChangeConnector: TaxCodeChangeCo
   }
 
   private def addEndDate(date: LocalDate, taxCodeRecord: TaxCodeRecord): TaxCodeRecordWithEndDate = {
+
+    val taxYear = TaxYear(TaxYearResolver.taxYearFor(date))
+
+    val startDate =
```

Comment:
I think the code here is a little misleading.

The function is called 'addEndDate' and one of its operations is to update the start date. This seems wrong.

You need a better name or a different function